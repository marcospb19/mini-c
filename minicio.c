#include <stdio.h>

int getint() {
    int ret;
    scanf("%d", &ret);
    return ret;
}

void putint(int v) {
    printf("%d ", v);
}

void putnewline() {
    printf("\n");
}
