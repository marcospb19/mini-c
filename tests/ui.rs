#![feature(custom_inner_attributes)]

use unindent::unindent;

macro_rules! test {
    ($input:expr) => {
        insta::assert_display_snapshot!($input)
    };
}

#[test]
fn file_1() {
    test!(unindent(
        r#"
        #include "minicio.h"

        int max(int a, int b) {
            if (a > b) {
                return a;
            } else {
                return b;
            }
        }

        int main() {
            int a = getint();
            int b = getint();
            int maximum = max(a, b);

            putint(maximum);
            putnewline();

            return 0;
        }
    "#
    ));
}

#[test]
fn file_2() {
    test!(unindent(
        r#"
        #include "minicio.h"

        int mod(int a, int b) {
            return a - (a / b) * b;
        }

        int gcd(int a, int b) {
            if (b == 0) {
                return a;
            } else {
                return gcd(b, mod(a, b));
            }
        }

        int main() {
            int a = getint();
            int b = getint();
            int divisor = gcd(a, b);

            putint(divisor);
            putnewline();

            return 0;
        }
    "#
    ));
}
