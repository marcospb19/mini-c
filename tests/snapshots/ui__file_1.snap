---
source: tests/ui.rs
expression: "unindent(r#\"\n        #include \"minicio.h\"\n\n        int max(int a, int b) {\n            if (a > b) {\n                return a;\n            } else {\n                return b;\n            }\n        }\n\n        int main() {\n            int a = getint();\n            int b = getint();\n            int maximum = max(a, b);\n\n            putint(maximum);\n            putnewline();\n\n            return 0;\n        }\n    \"#)"
---
#include "minicio.h"

int max(int a, int b) {
    if (a > b) {
        return a;
    } else {
        return b;
    }
}

int main() {
    int a = getint();
    int b = getint();
    int maximum = max(a, b);

    putint(maximum);
    putnewline();

    return 0;
}

